### PR TITLE
Add views to pagerank and fix pull algorithms

### PR DESCRIPTION
**Pull request by gurbinder533**

### CODE REVIEW

File: libgalois/src/analytics/pagerank/pagerank-push.cpp [hunk 1]
```diff
@@ -31,16 +31,21 @@ struct NodeResidual : katana::AtomicPODProperty<PRTy> {};
 
 using NodeData = std::tuple<NodeValue, NodeResidual>;
 using EdgeData = std::tuple<>;
-typedef katana::TypedPropertyGraph<NodeData, EdgeData> Graph;
-typedef typename Graph::Node GNode;
+// typedef katana::TypedPropertyGraph<NodeData, EdgeData> Graph;
+// typedef typename Graph::Node GNode;
+
+using Graph = katana::TypedPropertyGraphView<
```
Comment (uditagar) [inline libgalois/src/analytics/pagerank/pagerank-push.cpp]: I am thinking (even for other apps) that we should not create a default view, instead just use the full property graph. Default views somehow have very high overhead at the moment.

**Review by uditagar — APPROVED** [review]
everything looks fine, except Default view. This Default view is plaguing all apps (in which it is implemented).

File: libgalois/src/analytics/pagerank/pagerank-push.cpp (hunk 1 above)
Comment (gurbinder533) [reply libgalois/src/analytics/pagerank/pagerank-push.cpp]: We should definitely use default view but it should not have overhead. I did not measure it but why should it have any overhead. Did you measure it?

**Review by amazloumi — COMMENTED** [review]
Everything looks fine to me!

File: libgalois/src/analytics/pagerank/pagerank-push.cpp (hunk 1 above)
Comment (uditagar) [reply libgalois/src/analytics/pagerank/pagerank-push.cpp]: for louvain it takes extra 70 seconds with 32 threads on friendster. What I meant is to use a template `Graph`, which can be used to run our apps with views as well as with the full property graph. just `using Graph = ...` thing will change.

Comment (gurbinder533) [reply libgalois/src/analytics/pagerank/pagerank-push.cpp]: No, we should move everything to use views I feel. Mix and matching different types is not good.

**Pull request merged**